backend/src/test/java/com/woowacourse/levellog/authentication/application/OAuthServiceTest.java

```diff
@@ -0,0 +1,82 @@
+package com.woowacourse.levellog.authentication.application;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.when;
+
+import com.woowacourse.levellog.application.MemberService;
+import com.woowacourse.levellog.authentication.domain.GithubOAuthClient;
+import com.woowacourse.levellog.authentication.domain.JwtTokenProvider;
+import com.woowacourse.levellog.authentication.dto.GithubCodeRequest;
+import com.woowacourse.levellog.authentication.dto.GithubProfileResponse;
+import com.woowacourse.levellog.authentication.dto.LoginResponse;
+import com.woowacourse.levellog.domain.Member;
+import com.woowacourse.levellog.dto.MemberCreateDto;
+import java.util.Optional;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.transaction.annotation.Transactional;
+
+@SpringBootTest
+@Transactional
+@DisplayName("OAuthService의")
+class OAuthServiceTest {
+
+    @MockBean
+    private GithubOAuthClient githubOAuthClient;
+
+    @Autowired
+    private MemberService memberService;
+
+    @Autowired
+    private JwtTokenProvider jwtTokenProvider;
+
+    @Autowired
+    private OAuthService oAuthService;
+
+    @Nested
+    @DisplayName("login 메서드는")
+    class login {
+
+        @Test
+        @DisplayName("첫 로그인 시 회원가입하고 토큰과 이미지 URL를 반환한다.")
+        void loginFirst() {
+            // given
+            when(githubOAuthClient.getGithubAccessToken("githubCode")).thenReturn("accessToken");
+            when(githubOAuthClient.getGithubProfile("accessToken")).thenReturn(
+                    new GithubProfileResponse("12345", "로마", "imageUrl"));
```

Comment:
```suggestion
            BDDMockito.given(githubOAuthClient.getGithubAccessToken("githubCode"))
                    .willReturn("accessToken");
            BDDMockito.given(githubOAuthClient.getGithubProfile("accessToken"))
                    .willReturn(new GithubProfileResponse("12345", "로마", "imageUrl"));
```
Mockito.when 과 동일한 기능을 하는 BDDMockito.given이라는 메서드도 있어용 (내부 구현을 보시면 바로 아시겠지만 정말 메서드 이름 바꾸기만 해요)
저는 메서드 모킹은 주로 given 블록에 작성하는데 `when` 메서드를 호출하면 조금 어색하더라고용. 그래서 BDDMockito.given을 선호하는데 로마&페퍼는 어떠신가용

Reply:
좋은 의견 감사합니다~! 😃
페퍼와 이야기해 본 결과 `BDDMockito` 를 사용해서 `given` `willReturn` 으로 하는 것이 가독성이 좋겠다는 의견으로 모아져서 반영해 보도록 할게요!!